### DISCUSSION TITLE
Musicbrainz as a SongTag Service?

### DISCUSSION THREAD

**Discussion by hasezoey** [tramhao/termusic#283] [Ideas]
Seeing as there is a tageditor and some service provides, how about adding support for [`musicbrainz`](https://musicbrainz.org/)?

I dont really know the full specifications a service in the current tageditor in termusic needs, would `musicbrainz` suffice for this?

In addition to normal operation, using `musicbrainz` would benefit from reading the MBID tags on the files and also writing them.

There also exists a rust crate already, which seems to be reasonably working and maintained, namely `musicbrainz` crate: [`musicbrainz-rs`](https://docs.rs/musicbrainz_rs/0.5.0/musicbrainz_rs/index.html), though i have not tried it yet.

Notes:
- musicbrainz does not store covers / images themself, but via coverartarchive.org
- musicbrainz does not store lyrics, but links to other providers

PS: this is just a random idea that came to me, i do not personally use the termusic songtag / tageditor in any way

**Comment by tramhao** [comment]
For the lyrics, biggest problem is timestamp. If no timestamp, current lyric control can not show it. Spunds like a good idea. 
When I start termusic, the main usage is to download from youtube, then embed lyrics from netease server. Other services are added later to find more lyrics. If no lyric, we can still use other info, then fetch lyrics from old servers.
